File: backend/src/main/kotlin/ru/itmo/db/coursework/api/mapping/CircuitBoardInstanceDtoMapper.kt
package ru.itmo.db.coursework.api.mapping

import org.springframework.stereotype.Service
import ru.itmo.db.coursework.api.dto.CircuitBoardInstanceDto
import ru.itmo.db.coursework.model.CircuitBoardInstance

@Service
class CircuitBoardInstanceDtoMapper {
    fun fromDto(dto: CircuitBoardInstanceDto) = with(dto, {
        CircuitBoardInstance(
            modelId = modelId,
            amount = amount
        )
    })

    fun toDto(model: CircuitBoardInstance) = with(model, {
        CircuitBoardInstanceDto(
            modelId = modelId,
            amount = amount
        )
    })
}

File: frontend/src/api/mapping/CircuitBoardInstanceDtoMapper.ts
import TeaDto from "$src/api/dto/TeaDto";
import Tea from "$src/game/model/Tea";
import CircuitBoardInstanceDto from "$src/api/dto/CircuitBoardInstanceDto";
import CircuitBoardInstance from "$src/game/model/CircuitBoardInstance";

export default class CircuitBoardInstanceDtoMapper {
  public static fromDto(dto: CircuitBoardInstanceDto) {
    return new CircuitBoardInstance(dto.model_id, dto.amount)
  }

  public static toDto(model: CircuitBoardInstance) {
    return new CircuitBoardInstanceDto(model.modelId, model.amount)
  }
}